internal/util/askserver.go

```diff
@@ -0,0 +1,63 @@
+package util
+
+import (
+	"context"
+	"slices"
+
+	"github.com/10gen/migration-verifier/mslices"
+	"github.com/10gen/migration-verifier/option"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// ServerThinksTheseMatch runs an aggregation on the server that determines
+// whether the server thinks a & b are equal. This allows you, e.g., to
```

Comment:
The comment is not very accurate. It has a shortcut that compares the values in memory first.

Reply:
That’s an implementation, though. It seems trivially true that two binary-equivalent BSON values are ones that the server would also consider equivalent.